anycable-go/templates/deployment.yml

```diff
@@ -163,4 +163,7 @@ spec:
         terminationMessagePolicy: File
       dnsPolicy: ClusterFirst
       restartPolicy: Always
+      dnsConfig:
```

Comment:
imho we shouldn't be changing templates, it will be really hard to maintain any upstream changes
i've seen some other alternatives such as https://github.com/kubernetes-sigs/kustomize and others, which we should aim to try

IMHO we should only fork open source repos to make changes that will go to the upstream branch, or we will miss important updates

Reply:
I'm not sure if using kustomize would be a better solution.
It would basically discard the existing helm chart and we would need to build all configs again just because of a single change.
We should check it there's a good way to integrate both, like this article shows: https://testingclouds.wordpress.com/2018/07/20/844/

I haven't used any of those tools before, so I'm not sure it's the best way.

@adimania thoughts?

Reply:
Helm and Kustomize are different tools used to do template based deployments. The helm chart is officially supported by anycable but they don't use local DNS to connect to redis like we do.
So options are:
- clone this chart and modify it
- write your own helm chart (which is, in a way above method only)
- don't use templates and write the complete object
- use another method of creating template like Kustomize, create the actual template and maintain that.

The first option looks the best to me. Note that this repo has helm chart only. No other code.

Reply:
Based on your comments it makes sense to me to clone the repo.
It's minimal effort, it can get updates from the source repo, we would not need to rewrite all configs.

Although I see other efforts to integrate both, such as https://medium.com/devops-dudes/helm-is-not-enough-you-also-need-kustomize-82bae896816e and https://blog.container-solutions.com/using-helm-and-kustomize-to-build-more-declarative-kubernetes-workloads

@adimania can you help me understand if it's worth using both?
Perhaps we can start with only a cloned repo now and investigate how to integrate both later?

Reply:
@adimania I get your thought process, makes sense, but still, name of the files are confusing IMHO. I am not 100% familiar with helm and other templating engines as I said(only a high level understanding)

Is this a common nomeclature for helm files? templates.yml?

Reply:
we don't have templates.yml
The nomenclature is 100% standard. I have not changed any file names. Here is the reference: https://helm.sh/docs/topics/charts/

Reply:
@adimania sorry, I meant the `templates` directory

Reply:
@raivil, I do not recommend using both helm and kustomize together. The blog you mentioned, itself outlines the biggest downside
```
So this seems like a great way to customize upstream Helm charts but what am I missing out on by doing this? First, Helm is no longer controlling releases of manifests into the cluster. This means that you cannot use helm rollback or helm list or any of the helm release related commands to manage your deployments.
```
We'll be giving up key helm features and would create a mess. Currently, there is no good usecase to use both of these.

Reply:
@danielnc added codeowners

Reply:
@adimania add daniel as owner of codewoners file :)

Reply:
@raivil nah, i am good